### ISSUE TITLE
Playgrounds are no longer added to the project correctly

### ISSUE THREAD

**Issue by daltonclaybrook** [yonaskolb/XcodeGen#894]
It looks like the 2.16.0 release may have introduced a regression where playgrounds are no longer being added to the Xcode project correctly. Instead, they are being added like any other folder, and the playground `Contents.swift` file is being added as a target member.

**Comment by brentleyjones** [comment]
If I had to guess it would be from https://github.com/yonaskolb/XcodeGen/pull/862.

@KhaosT would you mind taking a look at it?

**Comment by KhaosT** [comment]
Yeah sure.

@daltonclaybrook so in this case we are talking about a Xcode project with a Swift playground bundled in there?

**Comment by daltonclaybrook** [comment]
@KhaosT Yah, that's right.

**Comment by KhaosT** [comment]
Thanks.

So the underlying issue is that `.playground` has a UTI of `file.playground` when it's actually a folder, and that causes the check at [here](https://github.com/yonaskolb/XcodeGen/blob/379e0feb1d9730fc16af8b6e707fc4768122cfc7/Sources/XcodeGenKit/XCProjExtensions.swift#L69) to fail.

There are two possible fixes we can go with, one is to add this UTI to folder special case so it can be added correctly. The other one is to partially change the behavior introduced in #826 to only allow folders with extension that's not mapped to a known UTI.

@brentleyjones do you have a preference?

**Comment by brentleyjones** [comment]
> The other one is to partially change the behavior introduced in #826 to only allow folders with extension that's not mapped to a known UTI.

That seems like the more wholistic change to me.

**Comment by KhaosT** [comment]
👍 The PR is up for review ^^

**Comment by KhaosT** [comment]
@daltonclaybrook quick question for you, when you add a Playground to Xcode, do you include it as part of a target? And if so, what do you expect when you build the app? Should the Swift playground get copied to the final target or not...?

**Comment by daltonclaybrook** [comment]
@KhaosT We do want the playground to be brought into the Xcode project along with the sources for a target, but that playground should not be a member of the target and should not be copied into the target's bundle. That being said, if there is a [setting under Sources](https://github.com/yonaskolb/XcodeGen/blob/master/Docs/ProjectSpec.md#sources) that we can specify to get this behavior, I'm happy to use it.

**Comment by KhaosT** [comment]
@daltonclaybrook Before you update to `2.16.0`, the Swift playground was not copied to the target right? Do you mind to share the segment of the configuration that included the playground?

**Comment by daltonclaybrook** [comment]
Yes, I believe the playground was not copied to the target pre `2.16.0`.  Here is the snippet that included the playground:
```
    SurveyKit:
        type: framework
        platform: iOS
        sources:
            - SurveyKit
        settings:
            groups: [ExtensionAPI]
        dependencies:
            - target: StyleKit_iOS
            - target: ArgonKit_iOS
            - target: ArgonCoreKit_iOS
```
The playground is located at `SurveyKit/Sources/SurveyKit.playground`

**Comment by KhaosT** [comment]
@daltonclaybrook Actually I tested with `v2.15.1` and the playground was bundled with the target... if your check the `YourApp.app/Frameworks/SurveyKit.framework/`, you can actually find `SurveyKit.playground` in there.

Given that's an unexpected behavior to you, do you mind to open a separate issue specially for what kind of resources should be copied 😅 I'll use this issue to address the regression.

**Comment by daltonclaybrook** [comment]
@KhaosT Ah, okay, that's very interesting. I don't think I'm prepared to say what the expected behavior necessarily _should_ be (Though I'd be curious to hear what @brentleyjones things). As I said, I don't mind tweaking the target settings to make sure the playground isn't copied into the bundle. I also don't mind opening another issue to continue the discussion, though.

**Comment by daltonclaybrook** [comment]
@KhaosT I just opened #897

**Comment by yonaskolb** [comment]
I can't imagine that anyone would want to add playgrounds to a bundle, so the default should be to not add it to any build phases. If someone wanted to do that they can always add a specific override and set the build phase
